### PR TITLE
VZ. 58 7.62x39mm rifles

### PR DESCRIPTION
**Pull request by DoctorBoomstick**
#### Summary
Content "adds the 7.62x39mm VZ. 58 P, VZ. 58 V, and VZ. 58 Tactical Sporter to the game."

#### Purpose of change
there was no real purpose for adding these guns, save for the fact that I do happen to be quite partial to a fine Czech VZ carbine, and wanted it to be in the game. Also, the Tactical Sporter is relatively common in the real world, with actual VZ originals being some what present in the US as well in the form of imports. Apart for that, this PR just aims at expanding the list of 7.62x39mm rifles, because one just can’t ever have enough guns when the Blob comes knocking.

#### Describe the solution
I have added, in no particular order, the VZ. 58 P, 58 V, and 58 Tactical Sporter rifles to the game, as well as their associated item spawns and magazines.
The VZ. 58 P is a standard automatic rifle that uses proprietary magazines. It is, in game terms, functionally the same as an AKM, save for being shorter, lighter, and overall smaller. The V model has a buttstock built into it, whilst the Tactical Sporter has a buttstock, muzzle break, and side rail.

#### Describe alternatives you've considered
possibly adding the IWI Galil rifle and associated variants first.

#### Testing
spawned in with each rifle and a collection of magazines, and went through systematically firing rounds off with each weapon, and testing the associated gun mods.

#### Additional context
[VZ 58 Tactical 762x39mm - AtlanticFirearms.com](https://atlanticfirearms.com/czech-small-arms-vz-58-tactical)
http://www.theballisticassistant.com/vz-58-purchase-the-parts-kit-and-receiver/
https://www.thetruthaboutguns.com/gun-review-czech-vzor-58/
https://www.youtube.com/watch?v=iCqElIDUzeI

### CODE REVIEW

File: data/json/itemgroups/Weapons_Mods_Ammo/guns.json [hunk 1]
```diff
@@ -424,6 +425,7 @@
       { "item": "ruger_mini", "prob": 1, "charges-min": 0, "charges-max": 5 },
       { "item": "sks", "prob": 1, "charges-min": 0, "charges-max": 10 },
       { "item": "win70", "prob": 2, "charges-min": 0, "charges-max": 3 },
+      { "item": "vz58ts", "prob": 10, "charges-min": 0, "charges-max": 30 },
```
Comment (anothersimulacrum) [inline data/json/itemgroups/Weapons_Mods_Ammo/guns.json:428]: This seems quite common compared to the other guns in this group.

File: data/json/items/gun/762.json [hunk 2]
```diff
@@ -236,5 +236,91 @@
...
+    "variant_type": "gun",
+    "variants": [
+      {
+        "id": "vz58_v",
+        "name": { "str": "Samopal VZ 58 V" },
+        "description": "This lightweight and relatively svelte cold war assault rifle was originally developed for use by the Czechoslovak army, in place of the various licenced Kalashnikov rifles which were then prevalent around eastern Europe.  Baring more than a passing resemblance to the AK-47, this weapon nevertheless features a completely different array of internals and possesses no parts interchangeability with its Russian counterpart.  Making use of proprietary magazines and a 15.4 inch barrel, the VZ. 58 is regarded by some to be the Kalashnikov’s lesser known superior.  Likely reconstructed from a pre-1986 imported parts kit, or brought back from a 3rd world war zone, this particular weapon features a removable side-folding buttstock and a 'V' suffixed designation, which both mark it as an original airborne or crew issued model.",
+        "weight": 1
+      }
+    ],
+    "built_in_mods": [ "folding_stock" ]
```
Comment (anothersimulacrum) [inline data/json/items/gun/762.json:304]: I might just do this as a variant of the above one, and have all the spawns on this variant also spawn with a folding stock (like the m38dmr).

**Comment by DoctorBoomstick** [comment]
@anothersimulacrum, with regards to the VZ. 58 V, the reason why I made it be its own thing was to create a distinction between it and the regularly issued gun, as they were two different firearms used for two different tasks. Though, I suppose that since the whole variants system exists for a situation just like this, it only makes sense to utilise it.
regards the commonality of the VZ when put up against other rifles, I got to say, even I don't know why I made it be that common. thanks for pointing it out, and it'll be fixed shortly.

File: data/json/items/gun/762.json [hunk 3]
```diff
@@ -236,5 +236,91 @@
...
     ],
     "pocket_data": [ { "pocket_type": "MAGAZINE_WELL", "item_restriction": [ "akmag10", "akmag20", "akmag30", "akmag40", "akdrum75" ] } ]
+  },
+  {
+    "id": "vz58_p",
+    "copy-from": "rifle_auto",
+    "looks_like": "ar15",
+    "type": "GUN",
+    "name": { "str": "VZ infantry carbine" },
+    "description": "A cold war era Czechoslovak assault rifle with a fixed wood impregnated plastic buttstock, likely constructed stateside from a pre-1986 imported parts kit or brought back from a 3rd world war zone.  Externally similar to the AK family of rifles, this weapon nevertheless only shares the former’s 7.62x39mm chambering.  Featuring no parts interchangeability with its Russian doppelganger, this firearm makes use of a completely different internal operating system, is far more svelte than its other Warsaw Pact counterparts, and feeds from proprietary box magazines.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/762.json:246]:
```suggestion
    "description": "A Cold War era Czech assault rifle with a fixed wood impregnated plastic buttstock, likely constructed stateside from a pre-1986 imported parts kit or brought back from a third world war zone.  Externally similar to the AK family of rifles, this weapon nevertheless only shares the former’s 7.62x39mm chambering.  Featuring no parts interchangeability with its Russian doppelganger, this firearm makes use of a completely different internal operating system, is far more svelte than its other Warsaw Pact counterparts, and feeds from proprietary box magazines.",
```

File: data/json/items/gun/762.json [hunk 4]
```diff
@@ -236,5 +236,91 @@
...
+    "variant_type": "gun",
+    "variants": [
+      {
+        "id": "vz58_p",
+        "name": { "str": "Samopal VZ 58 P" },
```
Comment (TheShadowFerret) [inline data/json/items/gun/762.json:251]:
```suggestion
        "name": { "str": "Samopal vz. 58 P" },
```

**Review by TheShadowFerret — COMMENTED** [review]
Just grammar fixes, as @anothersimulacrum already got the variant comment I was going to make.
1) ``Czechoslovak`` is only three letters from the full ``Czechoslovakian``
2) vz. 58 isn't supposed to be capitalized(I'd keep it caps for the generic name)
image
~Your first link from Atlantic Firearms
3) Cold War

Example of fix below, just apply to the others.

4) No stripper clips? ☹️

File: data/json/items/magazine/762.json [hunk 5]
```diff
@@ -102,5 +102,57 @@
...
     "flags": [ "MAG_BULKY" ],
     "pocket_data": [ { "pocket_type": "MAGAZINE", "rigid": true, "ammo_restriction": { "762": 75 } } ]
+  },
+  {
+    "id": "vz58mag10",
+    "looks_like": "stanag30",
+    "type": "MAGAZINE",
+    "name": { "str": "VZ 10-round magazine" },
+    "description": "A modified Czechoslovak rifle magazine which has been tweaked to only be capable of holding 10 rounds, in accordance with certain US magazine size restrictions.",
+    "variant_type": "gun",
```
Comment (TheShadowFerret) [inline data/json/items/magazine/762.json:112]:
```suggestion
    "description": "A modified Czechoslovak rifle magazine which has been tweaked to only be capable of holding ten rounds, in accordance with certain US magazine size restrictions.",
    "variant_type": "gun",
```
I don't think these magazines need a variant when there's nothing to confuse them with 😛

File: data/json/items/gun/762.json (hunk 2 above)
Comment (TheShadowFerret) [reply data/json/items/gun/762.json:304]: Also the folding stocks on these are made to be removable to easily swap between the regular and paratrooper variants, so I don't think these should be ``built_in``:
https://youtu.be/iCqElIDUzeI?t=450

**Comment by DoctorBoomstick** [comment]
@TheShadowFerret, Thanks as ever for the changes to the grammar and things of that nature, surprisingly difficult to catch those mistakenly uncapitalized/ capitalised letters when working with a screen reader. Now, few things here.
Re, stripper clips. Firstly, what I’ve seen points at the VZ being able to make use of standard SKS clips, which we’ve already got in the game. Secondly, to my knowledge at least, there is no current way to be able to use stripper clips on a rifle with an empty mag still inserted, like how they’d be used on real VZs. Essentially, as you might/ probably already know, IRL, you’d pull open your action and feed the clip’s rounds down into the magazine, thus removing the need of having to take the mag out. Again, far as I know you can’t do this with our guns in game yet, so adding clips is sort of a fruitless venture, and this is something that’ll be a factor in the future when I add things like Enfield rifles, which are also meant to use charger stripper clips to fill their own detachable mags. If this is something that you can do and that I’m just not aware of, that’d be great and I’d 100% look at fixing it accordingly.
Re, magazine names. Fair point, I made them be that way cause I had removed the, “58,” designation within the names of the actual rifles, so I wanted to keep consistency.
Re, the folding buttstock. Again, another fair point, I accidentally for some reason had a brain fart and forgot that that field doesn’t let the mods be removed  from the actual weapon.

**Comment by anothersimulacrum** [comment]
I think a variant is reasonable for the magazine names, fwiw.

File: data/json/items/gun/762.json [hunk 6]
```diff
@@ -236,5 +236,78 @@
...
+        "weight": 1
+      },
+      {
+        "id": "vz58_v",
+        "name": { "str": "Samopal VZ 58 V" },
+        "description": "This lightweight and relatively svelte cold war assault rifle was originally developed for use by the Czechoslovak army, in place of the various licenced Kalashnikov rifles which were then prevalent around eastern Europe.  Baring more than a passing resemblance to the AK-47, this weapon nevertheless features a completely different array of internals and possesses no parts interchangeability with its Russian counterpart.  Making use of proprietary magazines and a 15.4 inch barrel, the VZ. 58 is regarded by some to be the Kalashnikov’s lesser known superior.  Likely reconstructed from a pre-1986 imported parts kit, or brought back from a 3rd world war zone, this particular weapon features a removable side-folding buttstock and a 'V' suffixed designation, which both mark it as an original airborne or crew issued model."
```
Comment (anothersimulacrum) [inline data/json/items/gun/762.json:264]:
```suggestion
        "description": "This lightweight and relatively svelte Cold War assault rifle was originally developed for use by the Czechoslovakian army, in place of the various licensed Kalashnikov rifles which were then prevalent around eastern Europe.  Bearing more than a passing resemblance to the AK-47, this weapon nevertheless features a completely different array of internals and possesses no parts interchangeable with its Russian counterpart.  Making use of proprietary magazines and a 15.4 inch barrel, the VZ. 58 is regarded by some to be the Kalashnikov’s lesser known superior.  Likely reconstructed from a pre-1986 imported parts kit, or brought back from a 3rd world war zone, this particular weapon features a removable side-folding buttstock and a 'V' suffixed designation, which both mark it as an original airborne or crew issued model."
```

File: data/json/items/gun/762.json [hunk 7]
```diff
@@ -236,5 +236,78 @@
...
+    "//": "Couldn’t find exact price information on original VZ assault rifles, so did some guestimation based off of prices for similar Russian automatic weapon’s's parts kits.",
+    "price_postapoc": 4000,
+    "variant_type": "gun",
+    "variants": [
+      {
+        "id": "vz58_p",
+        "name": { "str": "Samopal vz. 58 P" },
+        "description": "This lightweight and relatively svelte Cold War assault rifle was originally developed for use by the Czechoslovakian army, in place of the various licenced Kalashnikov rifles which were then prevalent around eastern Europe.  Baring more than a passing resemblance to the AK-47, this weapon nevertheless features a completely different array of internals and possesses no parts interchangeability with its Russian counterpart.  Making use of proprietary magazines and a 15.4 inch barrel, the vz. 58 was regarded by some to be the Kalashnikov’s lesser known superior.  Likely reconstructed from a pre-1986 imported parts kit, or brought back from a 3rd world war zone, this particular weapon features a removable wood impregnated plastic buttstock and a 'P' suffixed designation, which both mark it as an original infantry model.",
```
Comment (anothersimulacrum) [inline data/json/items/gun/762.json:258]:
```suggestion
        "description": "This lightweight and relatively svelte Cold War assault rifle was originally developed for use by the Czechoslovakian army, in place of the various licensed Kalashnikov rifles which were then prevalent around eastern Europe.  Bearing more than a passing resemblance to the AK-47, this weapon nevertheless features a completely different array of internals and possesses no parts interchangeable with its Russian counterpart.  Making use of proprietary magazines and a 15.4 inch barrel, the vz. 58 was regarded by some to be the Kalashnikov’s lesser known superior.  Likely reconstructed from a pre-1986 imported parts kit, or brought back from a 3rd world war zone, this particular weapon features a removable wood impregnated plastic buttstock and a 'P' suffixed designation, which both mark it as an original infantry model.",
```

File: data/json/items/gun/762.json [hunk 8]
```diff
@@ -236,5 +236,78 @@
...
+    ],
+    "pocket_data": [ { "pocket_type": "MAGAZINE_WELL", "item_restriction": [ "vz58mag30", "vz58mag10" ] } ]
+  },
+  {
+    "id": "vz58ts",
+    "copy-from": "vz58_p",
+    "looks_like": "ar15",
+    "type": "GUN",
+    "name": { "str": "VZ tactical rifle" },
+    "description": "Inspired and built primarily  from older Czechoslovakian military combat rifles, this weapon bears an uncanny resemblance to the Kalashnikov despite sharing no components with it, including magazines.  Chambering the Soviet 7.62x39mm cartridge and featuring an American produced folding buttstock, receiver and barrel in order to have complied with US importation laws,, the weapon is equipped with a side mounted rail assembly as well as a detachable muzzle brake.  Despite the firearm’s Cold War design being nearly 80 years old, this modern iteration makes for a relatively portable and serviceable cataclysmic problem solver.",
```
Comment (anothersimulacrum) [inline data/json/items/gun/762.json:299]:
```suggestion
    "description": "Inspired and built primarily  from older Czechoslovakian military combat rifles, this weapon bears an uncanny resemblance to the Kalashnikov despite sharing no components with it, including magazines.  Chambering the Soviet 7.62x39mm cartridge and featuring an American produced folding buttstock, receiver and barrel in order to have complied with US importation laws, the weapon is equipped with a side mounted rail assembly as well as a detachable muzzle brake.  Despite the firearm’s Cold War design being nearly 80 years old, this modern iteration makes for a relatively portable and serviceable cataclysmic problem solver.",
```

File: data/json/items/gun/762.json [hunk 9]
```diff
@@ -236,5 +236,78 @@
...
     ],
     "pocket_data": [ { "pocket_type": "MAGAZINE_WELL", "item_restriction": [ "akmag10", "akmag20", "akmag30", "akmag40", "akdrum75" ] } ]
+  },
+  {
+    "id": "vz58_p",
+    "copy-from": "rifle_auto",
+    "looks_like": "ar15",
+    "type": "GUN",
+    "name": { "str": "VZ combat rifle" },
+    "description": "A Cold War era Czechoslovakian assault rifle, likely constructed stateside from a pre-1986 imported parts kit or brought back from a third world war zone.  Externally similar to the AK family of rifles, this weapon nevertheless only shares the former’s 7.62x39mm chambering.  Featuring no parts interchangeability with its Russian doppelganger, this firearm makes use of a completely different internal operating system, is far more svelte than its other Warsaw Pact counterparts, and feeds from proprietary box magazines.",
```
Comment (anothersimulacrum) [inline data/json/items/gun/762.json:246]:
```suggestion
    "description": "A Cold War era Czechoslovakian assault rifle, likely constructed stateside from a pre-1986 imported parts kit or brought back from a third world war zone.  Externally similar to the AK family of rifles, this weapon nevertheless only shares the former’s 7.62x39mm chambering.  Featuring no parts interchangeable with its Russian doppelganger, this firearm makes use of a completely different internal operating system, is far more svelte than its other Warsaw Pact counterparts, and feeds from proprietary box magazines.",
```

**Review by anothersimulacrum — APPROVED** [review]

**Comment by tinukedaya** [comment]
> 

Just want mention, that the 58 is actually a bit more important in the name than the "vz."

the "Sa vz.58" is short for "**Sa**mopal **vz**or 58" which can be translated as "Machine rifle model 58"  so using vz as a name for the gun makes not much sense.

**Comment by Zireael07** [comment]
@DoctorBoomstick: Not a Czech or Slovak native speaker but I'm almost certain "vz. 58" is parallel to Polish "wz 44", i.e. the letters stand for "wzór" (design) and are followed by the year the weapon was designed in. I know the pattern stands for both guns and bayonets.

File: data/json/items/gun/762.json [hunk 10]
```diff
@@ -236,5 +236,78 @@
...
+      [ "underbarrel mount", 1 ]
+    ],
+    "pocket_data": [ { "pocket_type": "MAGAZINE_WELL", "item_restriction": [ "vz58mag30", "vz58mag10" ] } ]
+  },
+  {
+    "id": "vz58ts",
+    "copy-from": "vz58_p",
+    "looks_like": "ar15",
+    "type": "GUN",
+    "name": { "str": "Czechpoint tactical rifle" },
```
Comment (anothersimulacrum) [inline data/json/items/gun/762.json:298]: This should include `vz. 58`, or all the variants should be called this (and the magazines should have variants to use this).

**Pull request merged**